xlsx_reader.py

```diff
@@ -0,0 +1,133 @@
+import xlrd
+from nameparser import HumanName
+from models import (Book, Author, Copy, Magazine, User)
+from init_db import db
+
+data = './biblioteka_probna.xlsx'     # please ensure if you have a proper file in folder
+workbook = xlrd.open_workbook(data)
+
+
+def get_or_create(session, model, **kwargs):    # checks if element exist, creates new if not
+    instance = session.query(model).filter_by(**kwargs).first()
+    if instance:
+        return instance
+    else:
+        instance = model(**kwargs)
+        session.add(instance)
+        session.commit()
+        return instance
+
+
+def get_author_name(authors):   # reads author's data from file
+    if (',' in authors and 'Jr.' not in authors) or (' and ' in authors) or ('&' in authors):
+        split_authors = authors.replace(' and ', ',').replace('&', ',').split(',')
+        ath = []
+        first_names = []
+        last_names = []
+
+        for auth in split_authors:
+            name = HumanName(str(auth))
+            first_name = (name.first + " " + name.middle).strip()
+            last_name = (name.last + " " + name.suffix).strip()
+            first_names.append(first_name)
+            last_names.append(last_name)
+            ath = list(zip(first_names, last_names))
```

Comment:
This `ath = list(zip(first_names, last_names))` should be invoked only once outside of `for`.